core/src/main/java/org/infinispan/configuration/cache/PersistenceConfiguration.java

```diff
@@ -117,6 +117,14 @@ public boolean usingAsyncStore() {
       return false;
    }
 
+   public boolean usingSegmentedStore() {
```

Comment:
I was just being consistent with the other method `usingAsyncStore`. I can change them both if you want.

Reply:
Actually I can't change the other, it has been that way for 6 years :) 

To be honest it would be best if it was `firstAdvancedLoaderIsSegment` or something like that, but is a bit convoluted. I will just clarify in comments that it returns for any store. Unfortunately there is no easy way to verify if a store is advanced loader or not either. I think the way it is would be fine.

Reply:
Ok, maybe the method here should stay as is, only with a javadoc saying it looks at any store. 

But shouldn't `PublisherManagerFactory` check only the first store instead? The first `AdvancedLoader` if really necessary, but I'd much rather simplify the logic so we always use the first loader for bulk operations and fail on start if there is an `AdvancedLoader` but the first one is a regular loader.

Reply:
Whoops, forgot the javadoc - I will add that. I might not have been clear earlier, but unfortunately there is no way to verify if the loader/stores are Advanced or not in the `PersistenceConfiguration` class.

Reply:
I don't want the check to be in the configuration (builder), I want it to be in `PersistenceManagerImpl`. The only thing I want to change is that instead of skipping all the non-`AdvancedLoader` loaders and returning the first `AdvancedLoader` when we need to iterate, we check on startup if there is any `AdvancedLoader` in the list and we throw a `CacheConfigurationException` if it's not the first (multiple `AdvancedLoader`s would still be allowed).

Reply:
Okay, we don't currently use a component for the sole purpose of configuration in any other use case, btw. 

Also it is perfectly fine if the first loader isn't `AdvancedLoader` for normal operation. I don't understand the reasoning you want this to throw an exception if the first one isn't advanced when more than one is present.

I can see the user wanting a simple Loader to be first since it may be faster than the actual `AdvancedLoader` but still want to support iteration. I would say we can discuss this in a separate issue as this is changing the behavior of how store ordering has worked since ISPN 6.

Reply:
Ok, let's discuss it separately.